extensions/typescript-language-features/src/extension.ts

```diff
@@ -42,6 +45,19 @@ export function activate(
 	const jsWalkthroughState = new JsWalkthroughState();
 	context.subscriptions.push(jsWalkthroughState);
 
+	let experimentTelemetryReporter: IExperimentationTelemetryReporter | null = null;
+	const packageInfo = getPackageInfo(context);
+	if (packageInfo) {
+		const { name: id, version, aiKey } = packageInfo;
+		const vscTelemetryReporter = new VsCodeTelemetryReporter(id, version, aiKey);
+		experimentTelemetryReporter = new ExperimentationTelemetryReporter(vscTelemetryReporter);
+		context.subscriptions.push(experimentTelemetryReporter);
+
+		// Don't bother creating an experimentation service if we don't have a valid reporter.
```

Comment:
Not sure what this comment means. Isn't the `ExperimentationService` always created?

Reply:
I have also made removed disposal from the `TelemetryReporter` in `extensions/typescript-language-features/src/utils/telemetry.ts` as well.